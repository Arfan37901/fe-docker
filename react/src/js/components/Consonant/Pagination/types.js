import {
    func,
    number,
} from 'prop-types';

export const PaginatorType = {
    onClick: func.isRequired,
    pageCount: number.isRequired,
    totalPages: number.isRequired,
    totalResults: number.isRequired,
    showItemsPerPage: number.isRequired,
    currentPageNumber: number.isRequired,
};

export const LoadMoreType = {
    show: number.isRequired,
    total: number.isRequired,
    onClick: func.isRequired,
};

export const RangeItemType = {
    item: number.isRequired,
    onClick: func.isRequired,
    currentPage: number.isRequired,
};
