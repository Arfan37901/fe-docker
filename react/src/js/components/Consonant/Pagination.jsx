import React from 'react';
import PropTypes from 'prop-types';

const Pagination = (props) => {
    const {
        pageCount,
        currentPageNumber,
        totalPages,
        onClick,
        showItemsPerPage,
        totalResults,
    } = props;

    /**
     * @function generateRange
     * @param {number} startVal - Start value in the range array;
     * @param {number} end - End value in the range array;
     * @return {Array}
     * @description Return range array of pages;
     */
    const generateRange = (startVal, end) => {
        let start = startVal;
        let step = 1;
        const range = [];

        if (end < start) {
            step = -step;
        }

        while (step > 0 ? end >= start : end <= start) {
            range.push(start);
            start += step;
        }

        // @TODO refactor, wrong logic in pagination;
        if (startVal !== 1 && end === totalPages && totalPages > pageCount) range.shift();

        return range;
    };

    /**
     * @function generatePageList
     * @param {number} pageCount - Total pages to display
     * @param {number} currentPageNumber - Current page user is on
     * @param {number} totalPages - Total number of pages available
     * @return {Array}
     * @description Return array of pages up to a max length of pageCount
     */
    const generatePageList = () => {
        const halfPageCount = Math.floor(pageCount / 2);
        let start;
        let end;


        if (totalPages <= (pageCount + 1)) {
            // show all pages
            start = 1;
            end = totalPages;
        } else {
            start = Math.min(
                Math.max(1, currentPageNumber - halfPageCount),
                totalPages - pageCount,
            );
            end = Math.max(
                Math.min(currentPageNumber + halfPageCount, totalPages),
                pageCount + 1,
            );
        }

        return generateRange(start, end);
    };

    const handleClick = (clickEvt) => {
        clickEvt.preventDefault();

        const { target } = clickEvt;
        let page;

        if (target.classList.contains('consonant-pagination--btn_prev')) {
            page = currentPageNumber - 1 > 0 ? currentPageNumber - 1 : 1;
        }
        if (target.classList.contains('consonant-pagination--btn_next')) {
            page = currentPageNumber + 1 < totalPages ? currentPageNumber + 1 : totalPages;
        }
        if (typeof page !== 'number') page = parseInt(target.firstChild.nodeValue, 10);

        onClick(page);
    };

    return (
        <div className="consonant-pagination">
            <div className="consonant-pagination--paginator">
                <button
                    data-testid="btn_prev"
                    onClick={handleClick}
                    type="buttton"
                    className="consonant-pagination--btn consonant-pagination--btn_prev">previous
                </button>
                <ul className="consonant-pagination--items">
                    {generatePageList().map(item => (
                        <li
                            key={item}
                            className={
                                currentPageNumber === item ?
                                    'consonant-pagination--item consonant-pagination--item_active' :
                                    'consonant-pagination--item'
                            }>
                            <button
                                data-testid="btn_page"
                                onClick={handleClick}
                                type="button"
                                className="consonant-pagination--item-btn">
                                {item}
                            </button>
                        </li>
                    ))}
                </ul>
                <button
                    data-testid="btn_next"
                    onClick={handleClick}
                    type="buttton"
                    className="consonant-pagination--btn consonant-pagination--btn_next">next
                </button>
            </div>
            <div data-testid="pagination--summary" className="consonant-pagination--summary">
                Showing
                <strong>
                    {`${currentPageNumber === 1 ? 1 : (currentPageNumber * showItemsPerPage) - (showItemsPerPage - 1)}-${currentPageNumber * showItemsPerPage}`} of
                </strong>
                {totalResults} Results
            </div>
        </div>
    );
};

export default Pagination;

Pagination.propTypes = {
    pageCount: PropTypes.number.isRequired,
    currentPageNumber: PropTypes.number.isRequired,
    totalPages: PropTypes.number.isRequired,
    onClick: PropTypes.func.isRequired,
    showItemsPerPage: PropTypes.number.isRequired,
    totalResults: PropTypes.number.isRequired,
};
