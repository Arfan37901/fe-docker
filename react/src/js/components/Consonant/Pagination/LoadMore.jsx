import React from 'react';
import { number, func } from 'prop-types';

import { useConfig } from '../Helpers/hooks';

const TLoadMore = {
    show: number.isRequired,
    total: number.isRequired,
    onClick: func.isRequired,
};

/**
 * Load More - Button That Naviates Users To The Next Page
 *
 * @component
 * @example
 * const props= {
    show: Int,
    total: Int,
    onClick: Function,
 * }
 * return (
 *   <LoadMore {...props}/>
 * )
 */
const LoadMore = ({
    show,
    total,
    onClick,
}) => {
    const getConfig = useConfig();

    /**
     * Authored Button Text
     * @type {String}
     */
    const loadMoreButtonText = getConfig('pagination', 'i18n.loadMore.btnText');

    /**
     * Authored Summary Text
     * @type {String}
     */
    const loadMoreQuantityText = getConfig('pagination', 'i18n.loadMore.resultsQuantityText');

    /**
     * Summary Of Load More Results To Show To Users
     * @type {String}
     */
    const summaryText = loadMoreQuantityText
        .replace('{start}', show)
        .replace('{end}', total);

    const shouldDisplayLoadMore = show > 0 && total > 0;
    const shouldDisplayLoadMoreBtn = show < total;

    return (shouldDisplayLoadMore) ? (
        <div
            data-testid="consonant-load-more"
            className="consonant-load-more">
            <div className="consonant-load-more--inner">
                <p
                    data-testid="consonant-load-more--text"
                    className="consonant-load-more--text">
                    {summaryText}
                </p>
                {shouldDisplayLoadMoreBtn &&
                    <button
                        type="button"
                        data-testid="load-more__button"
                        className="consonant-load-more--btn"
                        onClick={onClick}
                        tabIndex="0">
                        {loadMoreButtonText}
                    </button>
                }
            </div>
        </div>)
        : null;
};

LoadMore.propTypes = TLoadMore;

export default LoadMore;
