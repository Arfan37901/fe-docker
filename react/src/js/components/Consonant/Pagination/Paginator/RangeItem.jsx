import React, { useCallback } from 'react';
import classNames from 'classnames';

import { RangeItemType } from '../types';

/**
 * @component
 * @description Range item for Paginator component
 * @param {Object} props
 * @param {Number} props.item - Current reange item.
 * @param {Number} props.currentPage - Current selected page.
 * @param {Function} props.onClick - Click handler.
 */
const RangeItem = ({
    item,
    onClick,
    currentPage,
}) => {
    const className = classNames({
        'consonant-pagination--item': true,
        'consonant-pagination--item_active': currentPage === item,
    });

    const handleClick = useCallback((event) => {
        onClick(event, item);
    }, [item, onClick]);

    return (
        <li
            className={className}>
            <button
                tabIndex="0"
                type="button"
                onClick={handleClick}
                data-testid="btn_page"
                className="consonant-pagination--item-btn">
                {item}
            </button>
        </li>
    );
};

RangeItem.propTypes = RangeItemType;

export default RangeItem;
