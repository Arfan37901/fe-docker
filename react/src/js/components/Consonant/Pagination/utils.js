export const loadMoreSelector = ({
    pagination: {
        i18n: {
            loadMore: {
                btnText,
                resultsQuantityText,
            },
        },
    },
}) => ({
    loadMoreButtonText: btnText,
    loadMoreQuantityText: resultsQuantityText,
});

export const paginatorSelector = ({
    pagination: {
        i18n: {
            paginator: {
                prevLabel,
                nextLabel,
                resultsQuantityText,
            },
        },
    },
}) => ({
    nextLabel,
    prevLabel,
    quantityText: resultsQuantityText,
});
