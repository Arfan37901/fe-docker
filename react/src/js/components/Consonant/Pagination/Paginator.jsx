import React from 'react';
import PropTypes from 'prop-types';
import { useConfig } from '../../../utils/hooks';

const Paginator = (props) => {
    const {
        pageCount,
        currentPageNumber,
        totalPages,
        onClick,
        showItemsPerPage,
        totalResults,
    } = props;

    const getConfig = useConfig();

    const quantityText = getConfig('pagination', 'i18n.paginator.resultsQuantityText');
    const prevLabel = getConfig('pagination', 'i18n.paginator.prevLabel');
    const nextLabel = getConfig('pagination', 'i18n.paginator.nextLabel');

    /**
     * @function generateRange
     * @param {number} startVal - Start value in the range array;
     * @param {number} end - End value in the range array;
     * @return {Array}
     * @description Return range array of pages;
     */
    const generateRange = (startVal, end) => {
        let start = startVal;
        let step = 1;
        const range = [];

        if (end < start) {
            step = -step;
        }

        while (step > 0 ? end >= start : end <= start) {
            range.push(start);
            start += step;
        }

        // @TODO refactor, wrong logic in pagination;
        if (startVal !== 1 && end === totalPages && totalPages > pageCount) range.shift();

        return range;
    };

    /**
     * @function generatePageList
     * @param {number} pageCount - Total pages to display
     * @param {number} currentPageNumber - Current page user is on
     * @param {number} totalPages - Total number of pages available
     * @return {Array}
     * @description Return array of pages up to a max length of pageCount
     */
    const generatePageList = () => {
        const halfPageCount = Math.floor(pageCount / 2);
        let start;
        let end;


        if (totalPages <= (pageCount + 1)) {
            // show all pages
            start = 1;
            end = totalPages;
        } else {
            start = Math.min(
                Math.max(1, currentPageNumber - halfPageCount),
                totalPages - pageCount,
            );
            end = Math.max(
                Math.min(currentPageNumber + halfPageCount, totalPages),
                pageCount + 1,
            );
        }

        return generateRange(start, end);
    };

    const handleClick = (clickEvt) => {
        clickEvt.preventDefault();

        const { target } = clickEvt;
        let page;

        if (target.classList.contains('consonant-pagination--btn_prev')) {
            page = currentPageNumber - 1 > 0 ? currentPageNumber - 1 : 1;
        }
        if (target.classList.contains('consonant-pagination--btn_next')) {
            page = currentPageNumber + 1 < totalPages ? currentPageNumber + 1 : totalPages;
        }
        if (typeof page !== 'number') page = parseInt(target.firstChild.nodeValue, 10);

        onClick(page);
    };

    const getStartNumber = () => {
        if (currentPageNumber === 1) return 1;
        return (currentPageNumber * showItemsPerPage) - (showItemsPerPage - 1);
    };

    const getEndNumber = () => {
        const res = currentPageNumber * showItemsPerPage;
        return res < totalResults ? res : totalResults;
    };

    const renderQtyHTML = () => quantityText
        .replace('{start}', `<strong>${getStartNumber()}</strong>`)
        .replace('{end}', `<strong>${getEndNumber()}</strong>`)
        .replace('{total}', `<strong>${totalResults}</strong>`);

    return (
        <div className="consonant-pagination">
            <div className="consonant-pagination--paginator">
                <button
                    data-testid="btn_prev"
                    onClick={handleClick}
                    type="buttton"
                    className="consonant-pagination--btn consonant-pagination--btn_prev"
                    tabIndex="0">{prevLabel}
                </button>
                <ul className="consonant-pagination--items">
                    {generatePageList().map(item => (
                        <li
                            key={item}
                            className={
                                currentPageNumber === item ?
                                    'consonant-pagination--item consonant-pagination--item_active' :
                                    'consonant-pagination--item'
                            }>
                            <button
                                data-testid="btn_page"
                                onClick={handleClick}
                                type="button"
                                className="consonant-pagination--item-btn"
                                tabIndex="0">
                                {item}
                            </button>
                        </li>
                    ))}
                </ul>
                <button
                    data-testid="btn_next"
                    onClick={handleClick}
                    type="buttton"
                    className="consonant-pagination--btn consonant-pagination--btn_next"
                    tabIndex="0">{nextLabel}
                </button>
            </div>
            <div
                data-testid="pagination--summary"
                className="consonant-pagination--summary"
                dangerouslySetInnerHTML={{ __html: renderQtyHTML() }} />
        </div>
    );
};

export default Paginator;

Paginator.propTypes = {
    pageCount: PropTypes.number.isRequired,
    currentPageNumber: PropTypes.number.isRequired,
    totalPages: PropTypes.number.isRequired,
    onClick: PropTypes.func.isRequired,
    showItemsPerPage: PropTypes.number.isRequired,
    totalResults: PropTypes.number.isRequired,
};

Paginator.defaultProps = {
};
