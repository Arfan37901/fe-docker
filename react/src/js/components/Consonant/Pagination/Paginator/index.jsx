import React, { useMemo } from 'react';

import RangeItem from './RangeItem';
import { PaginatorType } from '../types';
import { paginatorSelector } from '../utils';
import { useConfigSelector } from '../../Helpers/hooks';
import {
    template,
    getEndNumber,
    generateRange,
    getStartNumber,
    getPageStartEnd,
} from '../../Helpers/general';

/**
 * Paginator - handles navigating between pages 1 to n for users
 *
 * @component
 * @example
 * const props= {
    pageCount: Number,
    currentPageNumber: Number,
    totalPages: Number,
    onClick: Function,
    showItemsPerPage: Boolean,
    totalResults: Number,
 * }
 * return (
 *   <Paginator {...props}/>
 * )
 */
const Paginator = ({
    onClick,
    pageCount,
    totalPages,
    totalResults,
    showItemsPerPage,
    currentPageNumber,
}) => {
    const {
        prevLabel,
        nextLabel,
        quantityText,
    } = useConfigSelector(paginatorSelector);

    const pageRange = useMemo(() => {
        /**
         * Start and end indexes of pages to build
         * @type {Number, Number}
         */
        const [pageStart, pageEnd] = getPageStartEnd(currentPageNumber, pageCount, totalPages);

        /**
         * Range of pages to build
         * @type {Array}
         */
        return generateRange(pageStart, pageEnd);
    }, [currentPageNumber, pageCount, totalPages]);


    const handleClickNext = (event) => {
        // ??
        event.preventDefault();

        const nextPageNotOutOfBounds = currentPageNumber + 1 < totalPages;
        const nextPage = nextPageNotOutOfBounds ? currentPageNumber + 1 : totalPages;

        onClick(nextPage);
    };

    const handleClickPrev = (event) => {
        // ??
        event.preventDefault();

        const nextPageNotNegative = currentPageNumber - 1 > 0;
        const prevPage = nextPageNotNegative ? currentPageNumber - 1 : 1;

        onClick(prevPage);
    };

    const handleClick = (event, nextPage) => {
        // ??
        event.preventDefault();

        onClick(nextPage);
    };

    /**
     * Summary Of Pagination Results
     * @type {String}
     */
    const paginationSummary = template(quantityText, {
        total: totalResults,
        start: getStartNumber(currentPageNumber, showItemsPerPage),
        end: getEndNumber(currentPageNumber, showItemsPerPage, totalResults),
    });

    return (
        <div
            className="consonant-pagination">
            <div
                className="consonant-pagination--paginator">
                <button
                    tabIndex="0"
                    type="button"
                    data-testid="btn_prev"
                    onClick={handleClickPrev}
                    className="consonant-pagination--btn consonant-pagination--btn_prev">
                    {prevLabel}
                </button>
                <ul
                    className="consonant-pagination--items">
                    {pageRange.map(item => (
                        <RangeItem
                            key={item}
                            item={item}
                            onClick={handleClick}
                            currentPage={currentPageNumber} />
                    ))}
                </ul>
                <button
                    tabIndex="0"
                    type="button"
                    data-testid="btn_next"
                    onClick={handleClickNext}
                    className="consonant-pagination--btn consonant-pagination--btn_next">
                    {nextLabel}
                </button>
            </div>
            <div
                data-testid="pagination--summary"
                className="consonant-pagination--summary">
                <strong>
                    {paginationSummary}
                </strong>
            </div>
        </div>
    );
};

Paginator.propTypes = PaginatorType;

export default Paginator;
