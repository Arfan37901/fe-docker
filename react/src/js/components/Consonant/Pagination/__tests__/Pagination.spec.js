import { screen, fireEvent, getNodeText } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Paginator from '../Paginator';

import { PAGE_LIST, PAGES_PROPS, DEFAULT_PROPS } from '../../Helpers/Testing/Constants/Pagination';

import makeSetup from '../../Helpers/Testing/Utils/Settings';
import { getItemsRange } from '../../Helpers/Testing/Utils/Pagination';

const setup = makeSetup(Paginator, DEFAULT_PROPS);

describe('Consonant/Pagination', () => {
    test('should render different items range', () => {
        PAGE_LIST.forEach((page) => {
            const { props: { showItemsPerPage, totalResults }, wrapper } =
                setup({ currentPageNumber: page });

            const itemRange = getItemsRange({ page, totalResults, itemsPerPage: showItemsPerPage });

            const paginationSummaryElement = screen.getByTestId('pagination--summary');

            expect(paginationSummaryElement).toHaveTextContent(itemRange);
            wrapper.unmount();
        });
    });

    describe('Check snapshots', () => {
        test('should renders correctly pagination range', () => {
            PAGES_PROPS.forEach((props) => {
                const { tree } = setup(props);

                expect(tree).toMatchSnapshot();
            });
        });
    });

    describe('Interaction with UI', () => {
        test('should call onClick', () => {
            const { props: { onClick } } = setup();

            const prevButton = screen.getByTestId('btn_prev');
            const nextButton = screen.getByTestId('btn_next');
            const pageButton = screen.queryAllByTestId('btn_page')[0];

            fireEvent.click(nextButton);

            expect(onClick).toHaveBeenCalledWith(2);

            fireEvent.click(prevButton);

            expect(onClick).toHaveBeenCalledWith(1);

            fireEvent.click(pageButton);
            const buttonText = getNodeText(pageButton);

            expect(onClick).toHaveBeenCalledWith(Number(buttonText));
        });
        test('should call onClick with 1', () => {
            const { props: { onClick } } = setup({ currentPageNumber: 0 });

            const prevButton = screen.getByTestId('btn_prev');

            fireEvent.click(prevButton);

            expect(onClick).toHaveBeenCalledWith(1);
        });
        test('should call onClick with totalPages', () => {
            const { props: { onClick } } = setup({ currentPageNumber: 2, totalPages: 1 });

            const nextButton = screen.getByTestId('btn_next');

            fireEvent.click(nextButton);

            expect(onClick).toHaveBeenCalledWith(1);
        });
        test('should call onClick with currentPageNumber - 1', () => {
            const { props: { onClick } } = setup({ currentPageNumber: 5 });

            const prevButton = screen.getByTestId('btn_prev');

            fireEvent.click(prevButton);

            expect(onClick).toHaveBeenCalledWith(4);
        });
        test('should call onClick not with 0', () => {
            const { props: { onClick } } = setup({ currentPageNumber: 1 });

            const prevButton = screen.getByTestId('btn_prev');

            fireEvent.click(prevButton);

            expect(onClick).toHaveBeenCalledWith(1);
        });
    });
});
