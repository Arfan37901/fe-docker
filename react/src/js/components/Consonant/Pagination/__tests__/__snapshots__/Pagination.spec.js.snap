// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Consonant/Pagination Check snapshots should renders correctly pagination range 1`] = `
<div
  className="consonant-pagination"
>
  <div
    className="consonant-pagination--paginator"
  >
    <button
      className="consonant-pagination--btn consonant-pagination--btn_prev"
      data-testid="btn_prev"
      onClick={[Function]}
      tabIndex="0"
      type="buttton"
    >
      Previous
    </button>
    <ul
      className="consonant-pagination--items"
    >
      <li
        className="consonant-pagination--item"
      >
        <button
          className="consonant-pagination--item-btn"
          data-testid="btn_page"
          onClick={[Function]}
          tabIndex="0"
          type="button"
        >
          1
        </button>
      </li>
    </ul>
    <button
      className="consonant-pagination--btn consonant-pagination--btn_next"
      data-testid="btn_next"
      onClick={[Function]}
      tabIndex="0"
      type="buttton"
    >
      Next
    </button>
  </div>
  <div
    className="consonant-pagination--summary"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Showing <strong>-9</strong>-<strong>0</strong> of <strong>100</strong> Results",
      }
    }
    data-testid="pagination--summary"
  />
</div>
`;

exports[`Consonant/Pagination Check snapshots should renders correctly pagination range 2`] = `
<div
  className="consonant-pagination"
>
  <div
    className="consonant-pagination--paginator"
  >
    <button
      className="consonant-pagination--btn consonant-pagination--btn_prev"
      data-testid="btn_prev"
      onClick={[Function]}
      tabIndex="0"
      type="buttton"
    >
      Previous
    </button>
    <ul
      className="consonant-pagination--items"
    >
      <li
        className="consonant-pagination--item consonant-pagination--item_active"
      >
        <button
          className="consonant-pagination--item-btn"
          data-testid="btn_page"
          onClick={[Function]}
          tabIndex="0"
          type="button"
        >
          1
        </button>
      </li>
      <li
        className="consonant-pagination--item"
      >
        <button
          className="consonant-pagination--item-btn"
          data-testid="btn_page"
          onClick={[Function]}
          tabIndex="0"
          type="button"
        >
          0
        </button>
      </li>
    </ul>
    <button
      className="consonant-pagination--btn consonant-pagination--btn_next"
      data-testid="btn_next"
      onClick={[Function]}
      tabIndex="0"
      type="buttton"
    >
      Next
    </button>
  </div>
  <div
    className="consonant-pagination--summary"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Showing <strong>1</strong>-<strong>10</strong> of <strong>100</strong> Results",
      }
    }
    data-testid="pagination--summary"
  />
</div>
`;

exports[`Consonant/Pagination Check snapshots should renders correctly pagination range 3`] = `
<div
  className="consonant-pagination"
>
  <div
    className="consonant-pagination--paginator"
  >
    <button
      className="consonant-pagination--btn consonant-pagination--btn_prev"
      data-testid="btn_prev"
      onClick={[Function]}
      tabIndex="0"
      type="buttton"
    >
      Previous
    </button>
    <ul
      className="consonant-pagination--items"
    >
      <li
        className="consonant-pagination--item"
      >
        <button
          className="consonant-pagination--item-btn"
          data-testid="btn_page"
          onClick={[Function]}
          tabIndex="0"
          type="button"
        >
          1
        </button>
      </li>
      <li
        className="consonant-pagination--item consonant-pagination--item_active"
      >
        <button
          className="consonant-pagination--item-btn"
          data-testid="btn_page"
          onClick={[Function]}
          tabIndex="0"
          type="button"
        >
          0
        </button>
      </li>
    </ul>
    <button
      className="consonant-pagination--btn consonant-pagination--btn_next"
      data-testid="btn_next"
      onClick={[Function]}
      tabIndex="0"
      type="buttton"
    >
      Next
    </button>
  </div>
  <div
    className="consonant-pagination--summary"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Showing <strong>-9</strong>-<strong>0</strong> of <strong>100</strong> Results",
      }
    }
    data-testid="pagination--summary"
  />
</div>
`;

exports[`Consonant/Pagination Check snapshots should renders correctly pagination range 4`] = `
<div
  className="consonant-pagination"
>
  <div
    className="consonant-pagination--paginator"
  >
    <button
      className="consonant-pagination--btn consonant-pagination--btn_prev"
      data-testid="btn_prev"
      onClick={[Function]}
      tabIndex="0"
      type="buttton"
    >
      Previous
    </button>
    <ul
      className="consonant-pagination--items"
    >
      <li
        className="consonant-pagination--item consonant-pagination--item_active"
      >
        <button
          className="consonant-pagination--item-btn"
          data-testid="btn_page"
          onClick={[Function]}
          tabIndex="0"
          type="button"
        >
          1
        </button>
      </li>
      <li
        className="consonant-pagination--item"
      >
        <button
          className="consonant-pagination--item-btn"
          data-testid="btn_page"
          onClick={[Function]}
          tabIndex="0"
          type="button"
        >
          0
        </button>
      </li>
    </ul>
    <button
      className="consonant-pagination--btn consonant-pagination--btn_next"
      data-testid="btn_next"
      onClick={[Function]}
      tabIndex="0"
      type="buttton"
    >
      Next
    </button>
  </div>
  <div
    className="consonant-pagination--summary"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Showing <strong>1</strong>-<strong>10</strong> of <strong>100</strong> Results",
      }
    }
    data-testid="pagination--summary"
  />
</div>
`;

exports[`Consonant/Pagination Check snapshots should renders correctly pagination range 5`] = `
<div
  className="consonant-pagination"
>
  <div
    className="consonant-pagination--paginator"
  >
    <button
      className="consonant-pagination--btn consonant-pagination--btn_prev"
      data-testid="btn_prev"
      onClick={[Function]}
      tabIndex="0"
      type="buttton"
    >
      Previous
    </button>
    <ul
      className="consonant-pagination--items"
    >
      <li
        className="consonant-pagination--item consonant-pagination--item_active"
      >
        <button
          className="consonant-pagination--item-btn"
          data-testid="btn_page"
          onClick={[Function]}
          tabIndex="0"
          type="button"
        >
          1
        </button>
      </li>
    </ul>
    <button
      className="consonant-pagination--btn consonant-pagination--btn_next"
      data-testid="btn_next"
      onClick={[Function]}
      tabIndex="0"
      type="buttton"
    >
      Next
    </button>
  </div>
  <div
    className="consonant-pagination--summary"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Showing <strong>1</strong>-<strong>10</strong> of <strong>100</strong> Results",
      }
    }
    data-testid="pagination--summary"
  />
</div>
`;

exports[`Consonant/Pagination Check snapshots should renders correctly pagination range 6`] = `
<div
  className="consonant-pagination"
>
  <div
    className="consonant-pagination--paginator"
  >
    <button
      className="consonant-pagination--btn consonant-pagination--btn_prev"
      data-testid="btn_prev"
      onClick={[Function]}
      tabIndex="0"
      type="buttton"
    >
      Previous
    </button>
    <ul
      className="consonant-pagination--items"
    >
      <li
        className="consonant-pagination--item"
      >
        <button
          className="consonant-pagination--item-btn"
          data-testid="btn_page"
          onClick={[Function]}
          tabIndex="0"
          type="button"
        >
          1
        </button>
      </li>
    </ul>
    <button
      className="consonant-pagination--btn consonant-pagination--btn_next"
      data-testid="btn_next"
      onClick={[Function]}
      tabIndex="0"
      type="buttton"
    >
      Next
    </button>
  </div>
  <div
    className="consonant-pagination--summary"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Showing <strong>-9</strong>-<strong>0</strong> of <strong>100</strong> Results",
      }
    }
    data-testid="pagination--summary"
  />
</div>
`;
